symphony-bdk-core/src/test/java/com/symphony/bdk/core/activity/command/SlashCommandTest.java

```diff
@@ -67,16 +77,136 @@ void testSlashCommandNotTriggered() {
+    cmd.setBotDisplayName(BOT_DISPLAY_NAME);
+    cmd.setBotUserId(BOT_USER_ID);
+    cmd.bindToRealTimeEventsSource(provider::setListener);
+
+    final String argValue = "value";
+    final V4MessageSent event = createMessageSentEvent(true, slashCommandName + " " + argValue);
+
+    provider.trigger(l -> l.onMessageSent(new V4Initiator(), event));
+
+    assertTrue(handlerCalled.get());
+    assertEquals(Collections.singletonMap(argName, argValue), arguments);
+  }
+
+  @Test
+  void testSlashCommandStringMentionCashtagHashtagArguments() {
+    final AtomicBoolean handlerCalled = new AtomicBoolean(false);
+    final Map<String, Object> arguments = new HashMap<>();
+
+    final Consumer<CommandContext> handler = c -> {
+      handlerCalled.set(true);
+      arguments.putAll(c.getArguments());
+    };
+
+    final RealTimeEventsProvider provider = new RealTimeEventsProvider();
+
+    final String slashCommandName = "/test";
+    String stringArgName = "stringArg";
+    String mentionArgName = "mentionArg";
+    String hashtagArgName = "hashtagArg";
+    String cashtagArgName = "cashtagArg";
+
+    final SlashCommand cmd = SlashCommand.slash(
+        slashCommandName + " {" + stringArgName + "} {" + mentionArgName + "} {" + hashtagArgName + "} {" + cashtagArgName + "}",
+        handler);
+    cmd.setBotDisplayName(BOT_DISPLAY_NAME);
+    cmd.setBotUserId(BOT_USER_ID);
+    cmd.bindToRealTimeEventsSource(provider::setListener);
+
+    final V4MessageSent event = createMessageSentEvent("<div><p>"
+            + "<span class=\"entity\" data-entity-id=\"0\">@" + BOT_DISPLAY_NAME + "</span>"
+            + slashCommandName + " "
+            + "value "
+            + "<span class=\"entity\" data-entity-id=\"1\">@John Doe</span> "
+            + "<span class=\"entity\" data-entity-id=\"2\">#myhashtag</span> "
+            + "<span class=\"entity\" data-entity-id=\"3\">$mycashtag</span>"
+            + "</p></div>",
+        "{\"0\":{\"id\":[{\"type\":\"com.symphony.user.userId\",\"value\":\"" + BOT_USER_ID + "\"}],\"type\":\"com.symphony.user.mention\"},"
+            + "\"1\":{\"id\":[{\"type\":\"com.symphony.user.userId\",\"value\":\"12345679\"}],\"type\":\"com.symphony.user.mention\"},"
+            + "\"2\":{\"id\":[{\"type\":\"org.symphonyoss.taxonomy.hashtag\",\"value\":\"myhashtag\"}],\"type\":\"org.symphonyoss.taxonomy\"},"
+            + "\"3\":{\"id\":[{\"type\":\"org.symphonyoss.fin.security.id.ticker\",\"value\":\"mycashtag\"}],\"type\":\"org.symphonyoss.fin.security\"}}");
+
+    provider.trigger(l -> l.onMessageSent(new V4Initiator(), event));
+
+    assertTrue(handlerCalled.get());
+    assertEquals(4, arguments.size());
+    assertEquals("value", arguments.get(stringArgName));
+    assertEquals(new Mention("@John Doe", 12345679L), arguments.get(mentionArgName));
+    assertEquals(new Hashtag("#myhashtag", "myhashtag"), arguments.get(hashtagArgName));
```

Comment:
now what it I try to do:
arguments.get(hashtagArgName) and expect to get a string out of it?

for instance if I don't really about the hashtag object but I'm fine to consider it a string

Reply:
We could have several methods like `getAsString(argName)` that would always return a string representation and specific methods like: `getMention(argName)` etc.